core/modules/ccontrol/testAntlr4GeneratedIR.cc

```diff
@@ -65,60 +66,52 @@ using namespace lsst::qserv;
 
 BOOST_AUTO_TEST_SUITE(Suite)
 
-/**
- * @brief Pusher is a set of recursive variadic functions to receive a variable number of arguments and push
- *        them onto a vector.
- *        This form is the last to be called when a series of args is being recursively pushed.
- * @tparam T The type contained by the vector.
- * @param vec The vector to push onto.
- * @param v The last object to be pushed.
- */
-template <typename T>
-void pusher(vector<T>& vec, T v) {
-    vec.push_back(v);
-}
-
 
 /**
  * @brief Pusher is a set of recursive variadic functions to receive a variable number of arguments and push
- *        them onto a vector.
+ *        them into a container that has a push_back function.
  *
- *        This form is used when explicitly decalring the template types when there is exactly one arg.
+ *        This form of `pusher` is the last to get called, for the single remaining element.
  *
- * @tparam T The type contained by the vector.
- * @tparam U The type of object being pushed.
- * @param vec The vector to push onto.
- * @param v The last object to be pushed.
+ * @tparam Container The type of container
+ * @tparam Types The template parameters the container accepts (e.g. for a vector it will be the element type
+ *               and the allocator type)
+ * @tparam T The type of the object to push into the container.
+ * @tparam TArgs The type of the remaining objects to push onto the vector (this should match T)
+ * @param container The container to push objects into.
+ * @param first The object to push into the container.
  */
-template <typename T, typename U>
-void pusher(vector<T>& vec, U v) {
-    vec.push_back(v);
+template <template <typename... Args> class Container, typename... Types, typename T, typename... TArgs>
```

Comment:
I'm not sure I understand why such complex template specification is needed in this context.